### DISCUSSION TITLE
一起写/开发者港湾（?

### DISCUSSION THREAD

**Discussion by BetaFoprhoton** [mouse0w0/MinecraftDeveloperGuide#87] [想法 / Ideas]
### 服务器（Server）：（愿景）
1. 服务器环境参照生存服务器，可以使用一些建筑所建筑的地图。
2. 提供聚集场所给开发者们，如聚集的广场大厅等
3. 加入一些mod，如农夫乐事，Create，网络唱片机，照相机和一些装饰性、休闲性的mod
4. 定期举行集体活动（枪械PVP，探索某mod维度，共同评鉴某人写的mod等），丰富开发者们的精神体验。
5. 聚集场所提供点子大厅、开发新手指引区、疑难技术解决提问区等区域，可将本repo（MinecraftDeveloperGuide）投影到服务器内
6. 聚集场所提供mod摊位，可以介绍开发者开发的mod、招募美术来为mod绘图等
7. 允许开发者自由建筑房屋。
8. 开发线上网页、让开发者可以访问服务器聊天/摸鱼、快速解决问题等。
9. 开发者可以将自己使用的IDE投影到服务器内，并提供公用调试环境（需申请，白名单使用）用于展示成果等。
10. 上文所述不限于Minecraft mod，其他相关内容开发者（HTML、C#、C++开发者等）也许也可以提供
### 模组（Mod）：
1. 将开发者使用的IDE投影到Minecraft中（参考[此mod](https://www.mcmod.cn/class/5200.html)，但也许还需加入投到大屏幕上的功能），需要开发对应IDE插件。
2. 将开发者的社区账号（Github，CSDN）以及个人开发信息映射到Minecraft内。
3. 提供悬浮按钮，材质包等
### 服务端插件（Server Plugin）
1. 提供GitHub聊天头饰前缀，服内GitHub浏览器等
### IDE插件（IDE Plugin）
1. 为jetbrains IDE / VScode写插件，使之能与Minecraft相连
## 困境 / 难点
1. **黑猴发布 / 群内开发者咕咕 / 人手急缺 /目前项目执行的开发者没有开服与活动执行专业经验 / 临近开学** 使得项目 **严重推迟**。
3. 开发者们的社恐使得推行可能**极为困难**。
4. 内存使用过大（IDE + Minecraft）可能导致一些内存小的电脑挂掉。
## 实现过程：（大概）
1. 搭建服务器、社区作为准备工作
2. 整合包选配、搭建服务器基本环境、服务器插件/mod开发
3. 社区/服务器宣传。
6. 基本维护（请隔壁（？））

**Comment by BetaFoprhoton** [comment]
和隔壁服务器的群友比较熟悉，他们也许有服务器可以借来一用（？

**Reply by kevin126ckw** [reply]
我有服务器或许可以拿来用

**Comment by BetaFoprhoton** [comment]
Feel sad :(

**Comment by killcerr** [comment]
支持（
活跃的社区需要活动（
悲，开学一周了）

**Comment by HappyRespawnanchor** [comment]
感觉是个好主意

**Comment by YakumoUUZ** [comment]
邮件已收到，幽幽玄代表八云之家向您问好。
